velox/exec/tests/HashJoinTest.cpp

```diff
@@ -4190,4 +4190,67 @@ TEST_F(HashJoinTest, spillFileSize) {
         .run();
   }
 }
+
+// The test is to verify if the hash build reservation has been released on task
+// error.
+DEBUG_ONLY_TEST_F(HashJoinTest, buildReservationReleaseCheck) {
+  std::vector<RowVectorPtr> probeVectors =
+      makeBatches(1, [&](int32_t /*unused*/) {
+        return std::dynamic_pointer_cast<RowVector>(
+            BatchMaker::createBatch(probeType_, 1000, *pool_));
+      });
+  std::vector<RowVectorPtr> buildVectors = makeBatches(10, [&](int32_t index) {
+    return std::dynamic_pointer_cast<RowVector>(
+        BatchMaker::createBatch(buildType_, 5000 * (1 + index), *pool_));
+  });
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  CursorParameters params;
```

Comment:
Any particular reason not to use AssertQueryBuilder?

Reply:
The test need to access the task to abort in the middle of a query execution? AssertQueryBuilder returns a velox task when the query completes.